x-pack/plugins/spaces/public/management/edit_space/customize_space/customize_space_avatar.tsx

```diff
@@ -106,138 +104,223 @@ export class CustomizeSpaceAvatar extends Component<Props, State> {
   public render() {
     const { space } = this.props;
 
-    const { initialsHasFocus, pendingInitials } = this.state;
-
-    const spaceColor = getSpaceColor(space);
-    const isInvalidSpaceColor = !isValidHex(spaceColor) && spaceColor !== '';
-
     return (
       <form onSubmit={() => false}>
         <EuiFormRow
           label={i18n.translate(
-            'xpack.spaces.management.customizeSpaceAvatar.initialItemsFormRowLabel',
+            'xpack.spaces.management.customizeSpaceAvatar.avatarTypeFormRowLabel',
             {
-              defaultMessage: 'Initials (2 max)',
+              defaultMessage: 'Avatar type',
             }
           )}
+          role="radiogroup"
+          aria-owns="avatar_type_initials avatar_type_image"
+          fullWidth
         >
-          <EuiFieldText
-            inputRef={this.initialsInputRef}
-            data-test-subj="spaceLetterInitial"
-            name="spaceInitials"
-            // allows input to be cleared or otherwise invalidated while user is editing the initials,
-            // without defaulting to the derived initials provided by `getSpaceInitials`
-            value={initialsHasFocus ? pendingInitials || '' : getSpaceInitials(space)}
-            onChange={this.onInitialsChange}
-            disabled={this.props.space.imageUrl && this.props.space.imageUrl !== '' ? true : false}
-          />
+          <EuiKeyPadMenu>
```

Comment:
Do we need to use KeyPadMenu here?  I think perhaps a selectable card is more typical for this type of usage.  And then we can remove the custom styling applied to it?

Reply:
I'm not sure whether the selectable card would work in this instance since it allows multi selection and feels more like it's own panel rather than a form element. 

There's an open issue to add this functionality to the keypad menu so the custom styling here is a stopgap solution: 
https://github.com/elastic/eui/issues/4000

The design for this is based on:
https://www.figma.com/proto/dBta1q3JgFe3Cfhw5h76Oq/%5BAwaiting-Dev%5D-Spaces-%26-Roles-Solution-Grouping?node-id=20%3A2087&scaling=min-zoom

Reply:
I'd like to jump in and make a suggestion here since the aim of this is to make UX improvements. When I look at the "Customize Avatar" section, the most prominent thing I see is the (most likely) one-time selection input of text vs image. I have to scan all the rest of the inputs to finally see the rendered Avatar. This Avatar needs to be the most prominent. I care about the final output... what does this look like?

My suggestions is to:
A. Move the rendered Avatar to the left side of the inputs (first instead of last).
B. While I do think this keypad item style of selection is a nice visual, the prominence is too much for this section. I would just not make them **have** to select between text and an image and just always show all inputs. Yes, the image will override the text and color, but I think that's a given knowing how avatars work. So that when they clear the uploaded image they just easily revert back to the text/color version and they don't have to re-select "Initials"

![Image 2021-05-27 at 9 13 00 AM](https://user-images.githubusercontent.com/549577/119833094-cdc86400-becc-11eb-9fdc-020d0a11b138.jpg)

Reply:
Sounds good @thomheymann — clearly i'm missing some discussions here :D  

Thanks for the suggestion @cchaos! Makes much more sense!

Reply:
@MichaelMarcialis How do you want to proceed with this design?

Reply:
Hey, gang! Thanks for the great feedback and discussion.

> When I look at the "Customize Avatar" section, the most prominent thing I see is the (most likely) one-time selection input of text vs image. I have to scan all the rest of the inputs to finally see the rendered Avatar. This Avatar needs to be the most prominent. I care about the final output... what does this look like?

> Move the rendered Avatar to the left side of the inputs (first instead of last).

I agree about my poor placement of the avatar preview in the original designs. I had initially placed the avatar to be prepended to the left of the inputs. However, as the avatar didn't have a label above to frame and align it with the following inputs, it felt floaty and out-of-place. I ended up placing it to the right because it felt better, but I fear I was too hasty in doing so, as you rightly point out that it loses a lot of prominence there. 

While I was tackling a similar form design in the my recent work on user settings, I switched to placing a larger sized avatar preview in the left column, beneath the form group description (see screenshots below). Perhaps the same approach can be taken here, as it would address the prominence while also avoiding the awkward placement issues mentioned above. Thoughts about applying similar avatar size and placement changes here?

<img width="786" alt="Avatar" src="https://user-images.githubusercontent.com/3884767/119878282-f0bc3d80-bef7-11eb-918e-486b7ec629c1.png">

<img width="786" alt="Avatar2" src="https://user-images.githubusercontent.com/3884767/119878290-f3b72e00-bef7-11eb-9cbf-9486ddec0ed7.png">

> While I do think this keypad item style of selection is a nice visual, the prominence is too much for this section. I would just not make them have to select between text and an image and just always show all inputs. Yes, the image will override the text and color, but I think that's a given knowing how avatars work. So that when they clear the uploaded image they just easily revert back to the text/color version and they don't have to re-select "Initials"

While I see your point, I have to respectfully disagree here. I'd prefer if we kept the current functionality of asking users to select initial/image mode and conditionally showing only those input elements that can be modified in the selected mode (especially if the above changes to the avatar size and placement are made, as prominence will be less of an issue). 

While I agree that it would be a relatively small cognitive load for users to come to the conclusion that applying an image will nullify any current or future initials and color values, I worry that it would feel unnecessarily messy. Also, any opportunity that we can save the user from having to think twice about an action is a win in my book.

Reply:
Have you also then thought about the unselected state? Meaning, when they've decided they want an image but haven't uploaded one yet. Or when they remove the uploaded image, what does the Avatar then render?

Reply:
Good question! Yes, both the older original designs and my more recent user settings designs have the avatar displayed in gray with a question mark in the case that the user is in image mode with no image supplied.

**Older original designs**
<img width="1152" alt="Frame 19" src="https://user-images.githubusercontent.com/3884767/120026472-2167aa00-bfc0-11eb-9dc7-f245f3fa83eb.png">

**Recent user settings designs (with new avatar positioning I suggest adopting here)**
<img width="786" alt="Avatar" src="https://user-images.githubusercontent.com/3884767/120026390-02691800-bfc0-11eb-823e-eb56a6d319b8.png">

Reply:
And that is the actually reason why I think it's just easier to fallback to the initials & color version when no image is selected. Otherwise there are several more clicks and I think someone could be confused as to why it's just a `?`. 

![image](https://user-images.githubusercontent.com/549577/120042818-d1e1a800-bfd8-11eb-94c3-653222cdab0c.png)

So clicking the `x` just reverts to the top screenshot (initials instead of a broken avatar).

And I'll now be quiet and walk away. Just wanted to share my 2 cents as to how it could be just a tiny bit easier/less clicks.

Reply:
I think allowing users to select what avatar type they want is clearer than showing all the different fields. I would find it confusing seeing the initials field when it has no effect on the avatar. Less clicks isn't necessarily better UX. 

Is this a blocker or can we merge this in? The PR addresses a couple of other bugs and accessibility issues and it would be great to get them merged in. I am happy to keep iterating on the design under a separate issue.

Reply:
Thanks for that feedback, @cchaos! I see what you're saying about the potential for additional clicks when using the initials/image mode selector. To be honest, I'm a bit torn on the best way to proceed, given the concerns with both directions. For sake of not stalling this PR further, I propose the following:

- Move the `EuiAvatar` preview to the left column, under the form description.
- Increase the `EuiAvatar` size via `size="xl"` prop.
- Keep the functionality of having the initials/image mode selector and monitor for user feedback (in case there is some confusion).
- Change the current `EuiKeyPadMenu` to an `EuiButtonGroup` for the initials/image mode selector. This will avoid having to use custom CSS, accounting for different styles for the different themes, or worrying about accessibility and semantics. Then, when the [new `EuiKeyPadMenu` enhancements](https://github.com/elastic/eui/issues/4000) are added into EUI, we can revisit and update (assuming no feedback regarding user confusion was given).

Do ya'll have any issues with this approach? Assuming all are good with this, would it be possible to make these changes, @thomheymann? And then @mdefazio can re-review.

Reply:
Just one last thing to mention is that, by default, EuiAvatar will still always render with a color, so transparent images will have this color behind it. Example: (I just grabbed you screenshot @MichaelMarcialis as a quick png).
<img width="487" alt="Screen Shot 2021-06-01 at 13 52 06 PM" src="https://user-images.githubusercontent.com/549577/120369062-cb0cab00-c2e0-11eb-866e-e23192281440.png">

Reply:
Thanks for the heads up. I think in the case of image-based avatars, we can probably just enforce a white background color (or perhaps transparent, if white isn't always desirable).

Reply:
Being able to specify a background colour might be desired when uploading a logo / icon as the avatar image so maybe that should always be selectable. 

We can of course change the behaviour to force a white or transparent background but this would be a Kibana wide change that could look strange in certain contexts. Transparent backgrounds would hide the shape of the space avatar which would make it difficult to recognise as such. Using a white background would pretty much do the same in light mode and look quite harsh in dark mode.

Reply:
> Being able to specify a background colour might be desired when uploading a logo / icon as the avatar image so maybe that should always be selectable.
> 
> We can of course change the behaviour to force a white or transparent background but this would be a Kibana wide change that could look strange in certain contexts. Transparent backgrounds would hide the shape of the space avatar which would make it difficult to recognise as such. Using a white background would pretty much do the same in light mode and look quite harsh in dark mode.

I agree -- this is closest to the current behavior as well, so our ux improvements won't end up surprising users

Reply:
> Being able to specify a background colour might be desired when uploading a logo / icon as the avatar image so maybe that should always be selectable.

Good point. Better to not adversely affect existing avatars or remove existing features/capabilities. I'm good with having the color picker in both initials and image modes. Let me know if you need any design support or if you feel comfortable making the changes directly.

Reply:
@MichaelMarcialis Could you please provide me with a design for these changes?

Reply:
@thomheymann: Certainly. I've put together a quick [updated mockup](https://www.figma.com/file/zWP1xc5TD1zVr4G77OxJaa/Space-Management-Update-Version-1-In-Progress?node-id=0%3A1) for the changes discussed above. Let me know if this works for your needs or if you have any questions. Thanks!

Reply:
Thanks Michael, I've updated the form accordingly.